internal/backend/runtime/omni/controllers/omni/kubernetes_status.go

```diff
@@ -240,6 +257,29 @@ func (ctrl *KubernetesStatusController) updateExposedServices(ctx context.Contex
 	return tracker.cleanup(ctx)
 }
 
+func (ctrl *KubernetesStatusController) buildExposedServiceURL(alias string) (string, error) {
+	apiURLParts := strings.SplitN(ctrl.advertisedAPIURL, "//", 2)
+	if len(apiURLParts) != 2 {
+		return "", fmt.Errorf("invalid advertised API URL protocol: %s", ctrl.advertisedAPIURL)
+	}
+
+	protocol := apiURLParts[0]
+	rest := apiURLParts[1]
+
+	restParts := strings.SplitN(rest, ".", 2)
+	if len(restParts) != 2 {
+		return "", fmt.Errorf("invalid advertised API URL: %s", ctrl.advertisedAPIURL)
+	}
+
+	instanceName := restParts[0]
+	rest = restParts[1]
+
+	// example: p-g3a4ana-demo.proxy-us.omni.siderolabs.io
+	url := protocol + "//" + workloadproxy.HostPrefix + "-" + alias + "-" + instanceName + "." + ctrl.workloadProxySubdomain + "." + rest
```

Comment:
Do we still use `p-` to route the requests?

Reply:
Actually, good point, we don't need it anymore. The instance name is still good to ensure there are no collisions, but `p-` is not needed really. Should we just drop it?

Reply:
Could be cool to provide a way to also make alias static, to get even prettier URL :)

Reply:
But I guess we should definitely drop `p-` prefix

Reply:
We had something like that as the initial idea back then, but there was a problem with potential collisions.

If it's ok, I'd leave that for another iteration to make sure we get it right if we do it again. There are some other topics we might want to address in a next iteration.

I'd say let's address the DNS propagation issue in this one. The changes we would add later would still use the new subdomain introduced here anyway.

But I'm gonna drop the `p-` - it's definitely not needed.

Reply:
dropped `p-`, also updated the omni-controller.